studio/components/layouts/ProjectLayout/ProjectLayout.tsx

```diff
@@ -72,20 +41,56 @@ const ProjectLayout: FC<Props> = ({
         {!hideIconBar && <NavigationBar />}
 
         {/* Product menu bar */}
-        {productMenu && !isLoading && (
+        <MenuBarWrapper isLoading={isLoading} productMenu={productMenu}>
           <ProductMenuBar title={product}>{productMenu}</ProductMenuBar>
-        )}
+        </MenuBarWrapper>
 
         <main
           style={{ maxHeight: '100vh' }}
           className="w-full flex flex-col flex-1 overflow-x-hidden"
         >
           {!hideHeader && <LayoutHeader />}
-          {renderContent()}
+          <ContentWrapper isLoading={isLoading}>{children}</ContentWrapper>
         </main>
       </div>
     </>
   )
 }
 
 export default observer(ProjectLayout)
+
+interface MenuBarWrapperProps {
+  isLoading: boolean
+  productMenu?: ReactNode
+}
+
+const MenuBarWrapper: FC<MenuBarWrapperProps> = observer(({ isLoading, productMenu, children }) => {
```

Comment:
i suppose this can actually live within <ProductMenuBar>?

other can we split this out into its own file? just for consistency